File: AddressBook/react-client/src/dtos/ContactEmail.ts
import Common from "../utility/Common";
import { EmailLabelRes } from "./EmailLabel";
import { PagedReq } from "./PagedReq";
import { ContactRes } from "./Contact";

export interface ContactEmailRes {
  contactEmailId?: string;
  email?: string;
  contactId?: string;
  contact?: ContactRes;
  emailLabelId?: string;
  emailLabel?: EmailLabelRes;
}

export class ContactEmailReqEdit {
  email?: string = "";
  contactId?: string = "";
  emailLabelId?: string = "";
  constructor(contactId?: string) {
    this.contactId = contactId;
  }
}

export class ContactEmailReqSearch extends PagedReq {
  contactId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {contactId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.contactId = contactId;
  }
}

File: AddressBook/Dtos/Dtos/ContactEmail.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using AddressBook.Common.Paging;

namespace AddressBook.Dtos
{
    public class ContactEmailRes
    {
        public int ContactEmailId { get; set; }
        public string? Email { get; set; }

        public int? ContactId { get; set; }
        public ContactRes? Contact { get; set; }
        public int? EmailLabelId { get; set; }
        public EmailLabelRes? EmailLabel { get; set; }
    }

    public class ContactEmailReqEdit
    {
        [Required]
        public string? Email { get; set; }

        public int? ContactId { get; set; }
        public int? EmailLabelId { get; set; }
    }

    public class ContactEmailReqSearch : PagedReq
    {
        public int? ContactId { get; set; }
    }
}
